templates/accounts.go

```diff
@@ -26,58 +26,126 @@ import (
 )
 
 const createAccountTemplate = `
-transaction(publicKeys: [[UInt8]], code: [UInt8]) {
+transaction(publicKeys: [[UInt8]]) {
   prepare(signer: AuthAccount) {
 	let acct = AuthAccount(payer: signer)
 
 	for key in publicKeys {
 	  acct.addPublicKey(key)
 	}
-	
-	if code.length > 0 {
-	  acct.setCode(code)
+  }
+}
+`
+
+const createAccountWitContractsTemplate = `
+transaction(publicKeys: [[UInt8]], contracts: {String: [UInt8]}) {
+  prepare(signer: AuthAccount) {
+	let acct = AuthAccount(payer: signer)
+
+	for key in publicKeys {
+	  acct.addPublicKey(key)
+	}
+
+    for contract in contracts.keys {
+	  acct.contracts.add(name: contract, code: contracts[contract]!)
 	}
   }
 }
 `
 
 // CreateAccount generates a transactions that creates a new account.
 //
-// This template accepts a list of public keys and a code argument, both of which are optional.
+// This template accepts an optional list of public keys.
 //
 // The final argument is the address of the account that will pay the account creation fee.
 // This account is added as a transaction authorizer and therefore must sign the resulting transaction.
-func CreateAccount(accountKeys []*flow.AccountKey, code []byte, payer flow.Address) *flow.Transaction {
+func CreateAccount(accountKeys []*flow.AccountKey, payer flow.Address) *flow.Transaction {
 	publicKeys := make([]cadence.Value, len(accountKeys))
 
 	for i, accountKey := range accountKeys {
 		publicKeys[i] = bytesToCadenceArray(accountKey.Encode())
 	}
 
 	cadencePublicKeys := cadence.NewArray(publicKeys)
-	cadenceCode := bytesToCadenceArray(code)
 
 	return flow.NewTransaction().
 		SetScript([]byte(createAccountTemplate)).
 		AddAuthorizer(payer).
+		AddRawArgument(jsoncdc.MustEncode(cadencePublicKeys))
+}
+
+// CreateAccountWithContracts generates a transactions that creates a new account.
+//
+// This template accepts a list of public keys and a contracts argument, both of which are optional.
+//
+// The contracts argument is a dictionary of *contract name*: *contract code (in bytes)*.
+// All of the contracts will be deployed to the account.
+//
+// The final argument is the address of the account that will pay the account creation fee.
+// This account is added as a transaction authorizer and therefore must sign the resulting transaction.
+func CreateAccountWithContracts(accountKeys []*flow.AccountKey, contracts map[string][]byte, payer flow.Address) *flow.Transaction {
```

Comment:
Do you it's worthwhile to have a separate function for this? I'm honestly not sure, but it would be possible to just modify `CreateAccount` and make the contracts argument optional. 

```go
func CreateAccount(accountKeys []*flow.AccountKey, contracts map[string][]byte, payer flow.Address)  { }
```

If `contracts` is `nil`, the loop below will just behave as if `contracts` is an empty map